cleaner/registry_docker.go

```diff
@@ -176,22 +178,47 @@ func (d DockerLocalRegistry) IsRegistryImagePresent() bool {
 	return false
 }
 
+func (d DockerLocalRegistry) IsImagePresent(name string) bool {
+
+	imageList, err := d.Connection.ImageList(context.Background(), types.ImageListOptions{})
+	if err != nil {
+		return false
+	}
+	for _, imagex := range imageList {
+		if imagex.RepoTags[0] == name || (imagex.RepoDigests != nil && strings.HasPrefix(imagex.RepoDigests[0], name)) {
+			return true
+		}
+	}
+	return false
+}
+
 func SetupDockerSocket() (DockerLocalRegistry, string, Docker) {
 	dockerSocketConn, err := GetDockerConnection()
+
 	if err != nil {
 		logrus.Errorf("Can't get Docker socket")
 		return DockerLocalRegistry{}, "", Docker{}
 	}
 	dockerSock := Docker{Connection: dockerSocketConn}
-	dockerSock.PurgeContainer("", REGISTRY)
+	logrus.Infof("Pull images for test")
+	err = dockerSock.PullImage(REGISTRY_FULLWITH_TAG)
+	if err != nil {
+		logrus.Errorf("Pull error in SetupDockerSocket %s", err)
+	}
+	_, err = dockerSock.PurgeContainer("", REGISTRY)
+	if err != nil {
+		logrus.Errorf("Purge  error in SetupDockerSocket %s", err)
+	}
 
 	d := DockerLocalRegistry{Connection: dockerSocketConn}
+	logrus.Info("Check if registry image is present :", d.IsRegistryImagePresent())
 	if !d.IsRegistryImagePresent() {
-		dockerSock.PullImage(TEST_IMAGE_TAG)
+		dockerSock.PullImage(REGISTRY_FULLWITH_TAG)
 	}
 	registryID := d.GetRegistryRunningID()
 	if len(registryID) == 0 {
```

Comment:
there are another place using a very similar `if, can't it be moved to a function to be reused?

Reply:
ok